templates/default/deploy-scripts/src/main.rs

```diff
@@ -0,0 +1,170 @@
+    #[error("anyhow error: {0}")]
+    AnyhowError(#[from] anyhow::Error),
+    #[error("There are unfinalized transactions. Transaction nonce is not reliable enough.")]
+    NonceNotFinal,
+    #[error("Transaction rejected: {0}")]
+    TransactionRejected(RPCError),
+    #[error("Transaction rejected: {0:?}")]
+    TransactionRejectedR(String),
+    #[error("Invalid block item: {0}")]
+    InvalidBlockItem(String),
+    #[error("Invalid contract name: {0}")]
+    InvalidContractName(String),
+    #[error("hex decoding error: {0}")]
+    HexDecodingError(#[from] FromHexError),
+    #[error("failed to parse receive name: {0}")]
+    FailedToParseReceiveName(String),
+    #[error("Json error: {0}")]
+    JSONError(#[from] serde_json::Error),
+    #[error("Parameter size error: {0}")]
+    ParameterSizeError(#[from] ExceedsParameterSize),
+    #[error("Receive name error: {0}")]
+    ReceiveNameError(#[from] NewReceiveNameError),
+    #[error("Contract name error: {0}")]
+    ContractNameError(#[from] NewContractNameError),
+    #[error("Reqwest error: {0}")]
+    ReqwestError(#[from] reqwest::Error),
+    #[error("Invalid metadata hash: {0}")]
+    InvalidHash(String),
+    #[error("IO error: {0}")]
+    IOError(#[from] std::io::Error),
+    #[error("Invoke contract failed: {0}")]
+    InvokeContractFailed(String),
+}
+
+/// Reads the wasm module from a given file path and file name.
+fn get_wasm_module(file: &Path) -> Result<WasmModule, DeployError> {
+    let wasm_module = std::fs::read(file).context("Could not read the WASM file")?;
+    let mut cursor = Cursor::new(wasm_module);
+    let wasm_module: WasmModule = concordium_rust_sdk::common::from_bytes(&mut cursor)?;
+    Ok(wasm_module)
+}
+
+/// Command line flags.
+#[derive(clap::Parser, Debug)]
+#[clap(author, version, about)]
+struct App {
+    #[clap(
+        long = "node",
+        default_value = "http://node.testnet.concordium.com:20000",
+        help = "V2 API of the Concordium node."
+    )]
+    url:      v2::Endpoint,
+    #[clap(
+        long = "account",
+        help = "Location path and file name of the Concordium account key file (e.g. \
+                ./myPath/3PXwJYYPf6fyVb4GJquxSZU8puxrHfzc4XogdMVot8MUQK53tW.export)."
+    )]
+    key_file: PathBuf,
+    #[clap(
+        long = "modules",
+        help = "Location paths and names of Concordium smart contract modules. Use this flag \
+                several times if you have several smart contract modules to be deployed (e.g. \
+                --modules ./myPath/default.wasm.v1 --modules ./default2.wasm.v1)."
+    )]
+    modules:  Vec<PathBuf>,
+}
+
+/// Main function: It deploys to chain all wasm modules from the command line
+/// `--modules` flags. Write your own custom deployment/initialization script in
+/// this function. An deployment/initialization script example is given in this
+/// function for the `default` smart contract.
+#[tokio::main(flavor = "current_thread")]
```

Comment:
Why do you choose this flavor over the defaults? Unless there is a reason to do it I would just use `tokio::main` without attributes.